Tests/PackageGraphTests/DependencyResolverTests.swift

```diff
@@ -376,6 +382,38 @@ class DependencyResolverTests: XCTestCase {
         }
     }
 
+    func testPrefetchContainers() throws {
+        // Test that resolver requests for needed containers.
+        let provider = MockPackagesProvider(containers: [
+            MockPackageContainer(name: "A", dependenciesByVersion: [v1: [
+                (container: "AA", versionRequirement: v1Range),
+                (container: "AB", versionRequirement: v1Range),
+            ]]),
+            MockPackageContainer(name: "AA", dependenciesByVersion: [v1: []]),
+            MockPackageContainer(name: "AB", dependenciesByVersion: [v1: []]),
+            MockPackageContainer(name: "B", dependenciesByVersion: [v1: []]),
+            MockPackageContainer(name: "C", dependenciesByVersion: [v1: [
+                (container: "CA", versionRequirement: v1Range),
+            ]]),
+            MockPackageContainer(name: "CA", dependenciesByVersion: [v1: [
+                (container: "CAA", versionRequirement: v1Range),
+                (container: "CAB", versionRequirement: v1Range),
+            ]]),
+            MockPackageContainer(name: "CAA", dependenciesByVersion: [v1: []]),
+            MockPackageContainer(name: "CAB", dependenciesByVersion: [v1: []]),
+        ])
+        let resolver = MockDependencyResolver(provider, MockResolverDelegate(), enablePrefetching: true)
+
+        let result = try resolver.resolve(constraints: [
+            MockPackageConstraint(container: "A", versionRequirement: v1Range),
+            MockPackageConstraint(container: "B", versionRequirement: v1Range),
+            MockPackageConstraint(container: "C", versionRequirement: v1Range),
+        ])
+
+        XCTAssertEqual(result[0].version, v1)
+        XCTAssertEqual(provider.requestedContainers, ["A", "B", "C", "A", "B", "C", "AA", "AB", "AA", "AB", "CA", "CA", "CAA", "CAB", "CAA", "CAB"])
```

Comment:
This test doesn't really seem to test what it is trying to -- sure it tests that the containers were requested in a particular order, but isn't what we really want to test that the resolver asked for containers before they were needed?

I'm not saying we should necessarily do the work to write such a test, but maybe it is worth simplifying this one if it doesn't really even cover the exact situation?